tests/Feature/Position/PositionTest.php

```diff
@@ -0,0 +1,174 @@
+                    ],
+                    [
+                        'id' => $positions->last()->id,
+                        'name' => $positions->last()->name,
+                        'frequency' => $positions->last()->frequency,
+                    ]
+                ]
+            ]
+        ]);
+    }
+
+    /* @test */
+    public function testItReturnsPositionById()
+    {
+        $positions = factory(Position::class, 2)->create();
+
+        $this->graphQL("
+        {
+            position(id: {$positions->first()->id}) {
+                id
+                name
+                frequency
+            }
+        }
+        ")->assertJson([
+            'data' => [
+                'position' => [
+                    'id' => $positions->first()->id,
+                    'name' => $positions->first()->name,
+                    'frequency' => $positions->first()->frequency,
+                ]
+            ]
+        ]);
+    }
+
+    /* @test */
+    public function testItReturnsPositionByCallsign()
+    {
+        $positions = factory(Position::class, 2)->create();
+
+        $this->graphQL("
+        {
+            position(callsign: \"{$positions->first()->callsign}\") {
+                id
+                name
+                frequency
+            }
+        }
+        ")->assertJson([
+            'data' => [
+                'position' => [
+                    'id' => $positions->first()->id,
+                    'name' => $positions->first()->name,
+                    'frequency' => $positions->first()->frequency,
+                ]
+            ]
+        ]);
+    }
+
+    /** @test */
+    public function testValidPositionCanBeCreated()
+    {
+        $this->graphQL('
+          mutation { 
+            createPosition(name: "Bristol Tower", callsign: "EGGD_TWR", frequency: "133.850", type: 4) {
+                id
+                name
+            }
+          }')->assertJsonStructure([
+            'data' => [
+                'createPosition' => [
+                    'id',
+                    'name'
+                ]
+            ]
+        ]);
+
+        $this->assertDatabaseHas('positions', [
+            'name' => 'Bristol Tower',
+            'callsign' => 'EGGD_TWR',
+            'frequency' => '133.850'
+        ]);
+    }
+
+    /** @test */
+    public function testDuplicatePositionCannotBeCreated()
+    {
+        factory(Position::class)->create([
+            'name' => 'Bristol Tower',
+            'callsign' => 'EGGD_TWR',
+            'frequency' => '133.850'
+        ]);
+
+        $this->graphQL('
+            mutation { 
+                createPosition(name: "Bristol Tower", callsign: "EGGD_TWR", frequency: "133.850", type: 4) {
+                    id
+                    name
+                }
+            }')->assertJsonFragment(["validation" => [
+                "callsign" => [
+                    "The callsign has already been taken."
```

Comment:
Can we do an `assertDatabaseMissing` to verify that the position has in fact not been created?

Reply:
I'm not sure if we can because we won't know what the PK is. I suppose we could just assert that the count of positions is only one?

Reply:
Yup could do!